商品削除機能

jodyyang007 opened the pull request:
# What
商品削除機能実装
# Why
商品削除機能実装

jodyyang007 committed:
商品削除機能

jodyyang007 commented:
ログイン状態の出品者のみ、詳細ページの削除ボタンを押すと、出品した商品を削除できる動画
https://gyazo.com/9d677a24d28c3d987e71a48f7652ca90
https://gyazo.com/e6c5bfb5241ea0b7e31f5fd101196051

maimai0401 commented on app/controllers/items_controller.rb:
```diff
@@ -38,6 +38,19 @@ def update
 end
 
+
+def destroy
```
◎出品者かどうかを判別する条件式がdestroyアクションにも適応されているか確認しましょう。

■理由
「「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す」という実装条件を満たす為です。

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。
コメントいたしましたので、ご確認をお願いいたします。
修正後、再度レビュー依頼フォームにてご提出くださいませ。

jodyyang007 committed:
出品者かどうかを判別する条件式がdestroyアクションにも適応されているか確認

def destroy
    if current_user == @item.user
    if @item.destroy
    redirect_to root_path
    else
      redirect_to item_path(@item)
    end
  else
    redirect_to item_path(@item)
  end
end

jodyyang007 commented:
「「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す」
https://gyazo.com/5020cae7c398b19f48df1c827abf311c

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。
LGTMとさせていただきます！実装お疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き実装頑張っていきましょう！
